### PR TITLE
Exportation Function 2TIFF for BioImg in TileDB Cloud

### PR DESCRIPTION
**Pull request by ktsitsi**
This PR:

- Adds a mirror functionality of ingestion API. The exportation uses in DAG's UDF nodes the `from_tiledb` API of `tiledb-bioimg` package to export tiledb arrays back to tiff files. The functionality has been requested by a customer. 
- Minor refactoring - isolating some helper functions

This PR has as dependence the change in tiledb-bioimg https://github.com/TileDB-Inc/TileDB-BioImaging/pull/91

### CODE REVIEW

**Review by Shelnutt2 — APPROVED** [review]

File: src/tiledb/cloud/bioimg/exportation.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+
+def export(
+    source: Union[Sequence[str], str],
+    output: str,
+    config: Mapping[str, Any],
+    *args: Any,
+    taskgraph_name: Optional[str] = None,
+    num_batches: Optional[int] = None,
+    threads: Optional[int] = 8,
```
Comment (thetorpedodog) [inline src/tiledb/cloud/bioimg/exportation.py:21]: nit: is `None` a valid value here? If not, this should just be `int`.

File: src/tiledb/cloud/bioimg/exportation.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    resources: Optional[Mapping[str, Any]] = None,
+    compute: bool = True,
+    namespace: Optional[str],
```
Comment (thetorpedodog) [inline src/tiledb/cloud/bioimg/exportation.py:24]: ` = None`?

File: src/tiledb/cloud/bioimg/exportation.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    if isinstance(source, str):
+        # Handle only lists
+        source = [source]
+
+    # Get the list of all BioImg samples input/out
+    samples = get_uris(source, output, config, "tiff")
+    batch_size, max_workers = scale_calc(source, num_batches)
+
+    # Build the task graph
+    dag_name = DEFAULT_DAG_NAME if taskgraph_name is None else taskgraph_name
```
Comment (thetorpedodog) [inline src/tiledb/cloud/bioimg/exportation.py:71]: `taskgraph_name or DEFAULT_DAG_NAME` does the same thing but shorter

**Review by thetorpedodog — APPROVED** [review]
A few minor things but overall looks good.

File: src/tiledb/cloud/bioimg/exportation.py (hunk 1 above)
Comment (ktsitsi) [reply src/tiledb/cloud/bioimg/exportation.py:21]: I will remove `threads` arg it is a placeholder for now, we better add it when truly functional.

File: src/tiledb/cloud/bioimg/exportation.py (hunk 2 above)
Comment (ktsitsi) [reply src/tiledb/cloud/bioimg/exportation.py:24]: Indeed DAG constructor allows `None` as default.

**Comment by thetorpedodog** [comment]
All looks good; will squash + merge once everything goes through.

**Pull request merged**